ext/snmp/snmp.c

```diff
@@ -701,18 +740,23 @@ static bool php_snmp_parse_oid(
 						}
 					}
 					if (idx_type < type_ht->nNumUsed) {
-						convert_to_string(tmp_type);
-						if (Z_STRLEN_P(tmp_type) != 1) {
+						zend_string *type = zval_try_get_string(tmp_type);
+						if (!type) {
+							php_free_objid_query(objid_query, oid_ht, value_str, value_ht, st);
+							return false;
+						}
+						if (ZSTR_LEN(type) != 1) {
 							zend_value_error("Type must be a single character");
-							efree(objid_query->vars);
+							zend_string_release(type);
```

Comment:
Sadly I can't make suggestions here, but you're leaking memory and making it too complicated.
Place this release above the if on line 748. Then, prior to releasing: read the first character and put it in a variable. Then remove `pptr = ZSTR_VAL(type);` and put the read character in line 755 by using the variable.